src/main/java/service/DataReaderImpl.java

```diff
@@ -0,0 +1,29 @@
+package service;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class DataReaderImpl implements DataReader {
+
+    @Override
+    public Map<Integer, Double> readPricesFromFile(String fileName) {
+        Map<Integer, Double> price = new HashMap<>();
+        File file = new File(fileName);
+
+        try {
+            Scanner scanner = new Scanner(file);
+            scanner.nextLine();
+            while (scanner.hasNextLine()) {
+                String[] lines = scanner.nextLine().split(",", 2);
+                price.put(Integer.parseInt(lines[0]), Double.parseDouble(lines[1]));
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("Файл " + fileName + " не найден!");
+            e.printStackTrace();
+        }
+        return price;
+    }
+}
```

Comment:
Please, try to do reading file to the map with Stream

Reply:
done